admin/schema.php

```diff
@@ -884,6 +884,10 @@
 
 # Release marker: 1.3.0
 
+# Per-bug access token for "public share URL" feature, #22203
+$g_upgrade[210] = array( 'AddColumnSQL', array( db_get_table( 'bug' ), "
```

Comment:
There are two models to come up with such tokens:

1. Use SHAxx on data include (issue_id + submission timestamp + global salt from config).  The global salt would be in $p_public_urls_salt config option for example.
2. Generate a crypto-secure token and store it for each issue.
3. Generate a crypto-secure token per share and have ability to revoke one but not the other.

The first approach doesn't require a schema change or upgrading of the data in the database, which is appealing. It doesn't allow revoking access to an issue though.  We could support revoked but it will require schema change to stamp an explicit token that overrides the SHA based approach for this specific issue (if we determine the revoke feature is needed in the future).

With the second issue, there is work for schema change + db upgrade, but there is the potential to revoke, which will revoke to all users.  Seems reasonable if we are going to use the revoke feature.

Third approach is likely an overkill.